linter/lints/cpcps/lint_crl_has_idp_subscriber_cert.go

```diff
@@ -25,25 +25,25 @@ Encrypt-specific suite of lints.
 func init() {
 	lint.RegisterRevocationListLint(&lint.RevocationListLint{
 		LintMetadata: lint.LintMetadata{
-			Name:          "e_crl_has_idp",
-			Description:   "Let's Encrypt issues sharded CRLs; therefore our CRLs must have an Issuing Distribution Point",
+			Name:          "e_crl_has_idp_subscriber_cert",
+			Description:   "Let's Encrypt Subscriber Cert CRLs must have the distributionPoint set",
 			Citation:      "",
 			Source:        lints.LetsEncryptCPS,
 			EffectiveDate: lints.CPSV33Date,
 		},
-		Lint: NewCrlHasIDP,
+		Lint: NewCrlHasIDPSubscriberCert,
 	})
 }
 
-func NewCrlHasIDP() lint.RevocationListLintInterface {
-	return &crlHasIDP{}
+func NewCrlHasIDPSubscriberCert() lint.RevocationListLintInterface {
+	return &crlHasIDPSubscriberCert{}
 }
 
-func (l *crlHasIDP) CheckApplies(c *x509.RevocationList) bool {
+func (l *crlHasIDPSubscriberCert) CheckApplies(c *x509.RevocationList) bool {
 	return true
 }
 
-func (l *crlHasIDP) Execute(c *x509.RevocationList) *lint.LintResult {
+func (l *crlHasIDPSubscriberCert) Execute(c *x509.RevocationList) *lint.LintResult {
```

Comment:
Similar idea here. I think we can get away with:
1) Extract the OnlyContainsUserCerts, OnlyContainsCACerts, and DistributionPoint fields
2) Assert that either:
   - just the OnlyContainsCACerts bool is set; or
   - both OnlyContainsUserCerts and DistributionPoint are set;
   - any other configuration is an error

I'm sorry that I led you astray in slack DMs. I wasn't thinking about the fact that lints need to know whether or not to run, and the only way for these lints to know whether or not to run is... to parse the whole IDP extension, at which point we might as well just do the check then and there. When I suggested two separate lints for the Subscriber and Subordinate CA paths here, I was forgetting that we can't inherently know which kind of CRL we're linting until we actually get to this parsing stage.

Reply:
Aha, field 8 was previously used and removed [here](https://github.com/letsencrypt/boulder/commit/7e994a12168aeaa4dda7c2ecc7ba92cc6c07e863#diff-80ea9561f1d738ba33d41a0776126deb77a355185a756cff4b77015bbd2a0e0f).